internal/cmd/wave/delete.go

```diff
@@ -112,53 +122,60 @@ func (x *CmdDelete) run(ctx context.Context) error {
 		return err
 	}
 
-	oceanClient, err := spotClient.Services().Ocean(x.opts.CloudProvider, spot.OrchestratorKubernetes)
-	if err != nil {
-		return err
-	}
-
-	c, err := oceanClient.GetCluster(ctx, x.opts.ClusterID)
+	waveClient, err := spotClient.Services().Wave()
 	if err != nil {
 		return err
 	}
 
-	// TODO Remove option to specify cluster-name on command line, or look up Ocean cluster by name,
-	// This will override the user supplied command line flag
-	x.opts.ClusterName = c.Name
-
-	// TODO Delete ocean cluster if it was provisioned
-	// TODO Delete kubernetes cluster if it was provisioned
-
-	if err := wave.ValidateClusterContext(c.Name); err != nil {
-		return fmt.Errorf("cluster context validation failure, %w", err)
+	if x.opts.Purge {
+		cluster, err := waveClient.GetCluster(ctx, x.opts.ClusterID)
+		if err != nil {
+			return err
+		}
+		if err := wave.ValidateClusterContext(cluster.Name); err != nil {
```

Comment:
we were thinking about removing this, since validation requires a local kubeconfig -- but we can refactor that later

Reply:
I have this `fs.BoolVar(&x.Purge, flags.FlagWaveDeleteClusterPurge, x.Purge, "delete all configuration (requires admin cluster access)"`

`purge` will remove the things that the deletion job doesn't remove (WaveEnvironment CRD, tide RBAC), and requires a local kubeconfig.

We can find some other solution for this.